packages/core/src/utilities/RLEVoxelMap.ts

```diff
@@ -0,0 +1,311 @@
+import { PixelDataTypedArray } from '../types';
+
+/**
+ * The RLERun specifies a contigous run of values for a row,
+ * where all indices (i only) from `[start,end)` have the specified
+ * value.
+ */
+export type RLERun<T> = {
+  value: T;
+  start: number;
+  end: number;
+};
+
+/**
+ * RLE based implementation of a voxel map.
+ * This can be used as single or multi-plane, as the underlying indexes are
+ * mapped to rows and hte rows are indexed started at 0 and continuing
+ * incrementing for all rows in the multi-plane voxel.
+ */
+export default class RLEVoxelMap<T> {
+  protected rows = new Map<number, RLERun<T>[]>();
```

Comment:
Could you kindly provide documentation for each of these properties, particularly for "rows"? It seems unfair that these properties lack documentation, while an obvious one like "numComps" does have it.